tools/halide_image_info.h

```diff
@@ -0,0 +1,306 @@
+// This header defines several methods useful for debugging programs that
+// operate on the Image class supporting images with arbitrary dimensions.
+//
+//   Image<uint16_t> input = load_image(argv[1]);
+//
+//   info(input, "input");  // Output the Image header info
+//   dump(input, "input");  // Dump the Image data
+//   stats(input, "input"); // Report statistics for the Image
+//
+//
+#ifndef HALIDE_TOOLS_IMAGE_INFO_H
+#define HALIDE_TOOLS_IMAGE_INFO_H
+
+#include <cassert>
+#include <cstdlib>
+#include <limits>
+#include <memory>
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+#include <cstdint>
+
+#include "HalideRuntime.h"
+
+namespace Halide {
+namespace Tools {
+
+static inline void print_dimid(int d, int val) {
+    static const char *dimid[] = {"x", "y", "z", "w"};
```

Comment:
Yes.  Arbitrary dimensions should work fine.  The first 4 will get textual names, anything beyond will be reported by index.